src/App.svelte

```diff
@@ -48,9 +54,35 @@
         }
     }
 
-    function filterMessages() {
-        filterPorts = filter.split('&&').map(Number);
-        messages = messages.filter((message) => !filterPorts.includes(message.dstPort));
+    function filterChangeHandler(): () => void {
+        return debounce(() => {
+            // TODO: validate queries as user types them.
+            console.log('filter is changing');
+            let parseResult = parseDSL(filter);
+            if (parseResult.lexErrors.length > 0 || parseResult.parseErrors.length > 0) {
+                console.log('Found some errors', parseResult.lexErrors, parseResult.parseErrors);
+                filterValid = false;
+                // TODO: highlight in red
+            } else {
+                filterValid = true;
+            }
+        }, 1000);
+    }
+
+    function applyFilter() {
+        console.log(`Going to parse ${filter}`);
+        let parseResult = parseDSL(filter);
```

Comment:
do you need to parse the filter every time we apply the filter?

Reply:
are you thinking of reusing `parseResult` from `filterChangeHandler()`?

Reply:
yes, or is there a reason to parse the filter again if it hasn't changed?

Reply:
Ah, right. No need for now then. We should reconsider it once we get significant amounts of events. Can you create a ticket for it?

Reply:
Sure, created https://github.com/honeynet/ochi/issues/49